### PR TITLE
UpdateOrCreateToken get secrets err handling optimization

### PR DESCRIPTION
**Pull request by helen-frank**
Signed-off-by: helen <haitao.zhang@daocloud.io>

**What type of PR is this?**
/kind feature

**What this PR does / why we need it**:
when UpdateOrCreateToken get secrets, if the error is not a NotFound, the error is thrown in time
**Which issue(s) this PR fixes**:
Fixes #

**Special notes for your reviewer**:

**Does this PR introduce a user-facing change?**:

```release-note
NONE
```

### CODE REVIEW

File: pkg/karmadactl/util/bootstraptoken/bootstraptoken.go [hunk 1]
```diff
@@ -320,6 +321,8 @@ func UpdateOrCreateToken(client kubeclient.Interface, failIfExists bool, token *
 	secret, err := client.CoreV1().Secrets(metav1.NamespaceSystem).Get(context.TODO(), secretName, metav1.GetOptions{})
 	if secret != nil && err == nil && failIfExists {
 		return fmt.Errorf("a token with id %q already exists", token.Token.ID)
+	} else if err != nil && !apierrors.IsNotFound(err) {
```
Comment (lonelyCZ) [inline pkg/karmadactl/util/bootstraptoken/bootstraptoken.go:324]: The other errors will be checked in `cmdutil.CreateOrUpdateSecret()`, it seems only preflight for `karmadactl token create` command and not to influence the CreateOrUpdateSecret phase.

And it refers to https://github.com/kubernetes/kubernetes/blob/cde45fb161c5a4bfa7cfe45dfd814f6cc95433f7/cmd/kubeadm/app/phases/bootstraptoken/node/token.go#L38

Comment (helen-frank) [reply pkg/karmadactl/util/bootstraptoken/bootstraptoken.go:324]: Yes, so I think this is an optimization feature rather than a bug.

Comment (helen-frank) [reply pkg/karmadactl/util/bootstraptoken/bootstraptoken.go]: @lonelyCZ https://github.com/kubernetes/kubernetes/pull/115000 kubeadm someone approved of my idea

File: pkg/karmadactl/util/bootstraptoken/bootstraptoken.go [hunk 2]
```diff
@@ -318,7 +319,10 @@ func CreateNewToken(client kubeclient.Interface, token *BootstrapToken) error {
 func UpdateOrCreateToken(client kubeclient.Interface, failIfExists bool, token *BootstrapToken) error {
 	secretName := bootstraputil.BootstrapTokenSecretName(token.Token.ID)
 	secret, err := client.CoreV1().Secrets(metav1.NamespaceSystem).Get(context.TODO(), secretName, metav1.GetOptions{})
-	if secret != nil && err == nil && failIfExists {
+	if err != nil && !apierrors.IsNotFound(err) {
+		return err
+	}
```
Comment (lonelyCZ) [inline pkg/karmadactl/util/bootstraptoken/bootstraptoken.go:324]: Great, this is also my point, it is clearer.

File: pkg/karmadactl/util/bootstraptoken/bootstraptoken.go (hunk 1 above)
Comment (lonelyCZ) [reply pkg/karmadactl/util/bootstraptoken/bootstraptoken.go]: Right, if the err is not nil, the `already exist` error will be skipped that could cause the unexpected result. This patch make it safer.

**Review by RainbowMango — APPROVED** [review]
/remove-kind feature
/kind cleanup
/approve

**Pull request merged**